### PR TITLE
Fixed-bug-LVJAVS-581

### PR DESCRIPTION
**Pull request by hrybs**
Fixed Bug LVJAVS-581
http://ssu-jira.softserveinc.com/browse/LVJAVS-581

### CODE REVIEW

File: src/main/webapp/resources/js/recurrent-cancel-validator.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
+
+/**
+ * Created by Sviatoslav Hryb on 29-Nov-16.
+ */
+
+var MAX_POSSIBLE_DATE = new Date(884541340800000); //Sat, 01 Jan 30000 00:00:00 GMT
```
Comment (andryspo) [inline src/main/webapp/resources/js/recurrent-cancel-validator.js]: use constant for this number

File: src/main/webapp/resources/js/recurrent-cancel-validator.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+ * Checks if the startDate or the endDate is not null or undefined
+ *
+ * @param startDate start of the date for checking
+ * @param endDate end of the day for checking
+ * @returns {boolean} if startDate or endDate is null or undefined - returns false.
+ * Otherwise  - returns true.
+ */
+function isNotEmptyDates(startDate, endDate) {
+    var result = true;
+    if (typeof startDate === 'undefined' || startDate === null){
```
Comment (andryspo) [inline src/main/webapp/resources/js/recurrent-cancel-validator.js]: null == undefined; you could check only one of this statements

File: src/main/webapp/resources/js/recurrent-cancel-validator.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    var result = true;
+    var arStartEv;
+    var i, event;
+    for (i = 0; i < allEvents.length; i++)
+        if (allEvents[i].recurrentId === recurrentId) {
+            event = allEvents[i];
+            break;
+        }
+    arStartEv = event.start.split('T');
+
```
Comment (andryspo) [inline src/main/webapp/resources/js/recurrent-cancel-validator.js]: what does 'T' mean?

Comment (hrybs) [inline src/main/webapp/resources/js/recurrent-cancel-validator.js]: Java Script string Constant for UTC time

File: src/main/webapp/resources/js/recurrent-cancel-validator.js (hunk 2 above)
Comment (hrybs) [inline src/main/webapp/resources/js/recurrent-cancel-validator.js]: You are right.

File: src/main/webapp/resources/js/recurrent-cancel-validator.js (hunk 1 above)
Comment (andryspo) [inline src/main/webapp/resources/js/recurrent-cancel-validator.js]: ok

**Pull request merged**